### PR TITLE
Fixing serum client go swaps

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: provider/common.go [hunk 1]
```diff
@@ -85,31 +88,41 @@ func buildBatchRequest(transactions interface{}, privateKey solana.PrivateKey, o
...
-				IsCleanup: txMsg.IsCleanup,
+				return nil, err
 			}
-		}
 
-		batchRequest.Entries = append(batchRequest.Entries, &oneRequest)
+			batchRequest.Entries = append(batchRequest.Entries, oneRequest)
+		}
+	default:
+		log.Fatalf("transactions has unknown type : %v", reflect.TypeOf(transactions))
```
Comment (kdonthi) [inline provider/common.go:111]: Can we return an error here instead of logging it?

**Review by kdonthi — APPROVED** [review]
Generally looks good, just the comment about returning the error instead of logging it.

Also, why is the orderbook stream not working in the integration tests?

File: provider/common.go (hunk 1 above)
Comment (tinystarinagalaxy) [reply provider/common.go:111]: fatal is better, we will catch the error early on in case. its fail fast and hard principle

**Comment by tinystarinagalaxy** [comment]
> Generally looks good, just the comment about returning the error instead of logging it.
> 
> Also, why is the orderbook stream not working in the integration tests?

not sure

**Comment by tinystarinagalaxy** [comment]
> Details

its failing on develop as well


INFO[2022-11-01 06:27:41] starting orderbook stream                    
INFO[2022-11-01 06:27:41] subscription error: rpc error: code = Unknown desc = provided market name/address was not found 
INFO[2022-11-01 06:27:42] response 1 received, data slot:158605149  orderbook:{market:"SOL/USDC"  marketAddress:"9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT"  bids:{price:32.87  size:19.6}  bids:{price:32.872  size:0.1}  bids:{price:32.88  size:9.8}  asks:{price:33.003  size:300}  asks:{price:33.004  size:869.3}  asks:{price:33.035  size:900}}  
INFO[2022-11-01 06:27:42] response 2 received, data slot:158605150  orderbook:{market:"SOL/USDC"  marketAddress:"9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT"  bids:{price:32.87  size:19.6}  bids:{price:32.872  size:0.1}  bids:{price:32.88  size:9.8}  asks:{price:33.003  size:300}  asks:{price:33.004  size:869.3}  asks:{price:33.035  size:900}}  
INFO[2022-11-01 06:27:42] response 3 received, data slot:158605150  orderbook:{market:"SOL/USDC"  marketAddress:"9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT"  bids:{price:32.87  size:19.6}  bids:{price:32.872  size:0.1}  bids:{price:32.88  size:9.8}  asks:{price:33.003  size:300}  asks:{price:33.004  size:869.3}  asks:{price:33.035  size:900}}  
INFO[2022-11-01 06:27:42] response 4 received, data slot:158605151  orderbook:{market:"SOL/USDC"  marketAddress:"9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT"  bids:{price:32.87  size:19.6}  bids:{price:32.872  size:0.1}  bids:{price:32.88  size:9.8}  asks:{price:33.003  size:300}  asks:{price:33.004  size:869.3}  asks:{price:33.035  size:900}}  
INFO[2022-11-01 06:27:42] response 5 received, data slot:158605151  orderbook:{market:"SOL/USDC"  marketAddress:"9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT"  bids:{price:32.87  size:19.6}  bids:{price:32.872  size:0.1}  bids:{price:32.88  size:9.8}  asks:{price:33.003  size:300}  asks:{price:33.004  size:869.3}  asks:{price:33.035  size:900}}  
INFO[2022-11-01 06:27:42] starting get prices stream                   
INFO[2022-11-01 06:27:42] skipping trades due to config                
FATA[2022-11-01 06:27:42] one or multiple examples failed

File: provider/common.go (hunk 1 above)
Comment (aspin) [reply provider/common.go:111]: best not enable this kind of behavior at all I think –– https://bloxroute.atlassian.net/browse/TRAD-285 for removing this completely

**Pull request merged**